custom/ilsgateway/tanzania/reports/alerts.py

```diff
@@ -19,16 +20,24 @@ class AlertReport(GenericTabularReport, CustomProjectReport, ProjectReportParame
     base_template = 'ilsgateway/base_template.html'
 
     @property
+    def sql_location(self):
+        location_id = self.request.GET.get('location_id')
+        if location_id:
+            return get_object_or_404(SQLLocation, location_id=location_id, domain=self.domain)
+        return get_object_or_404(SQLLocation, location_type__name='MOHSW', domain=self.domain)
+
+    @property
     def headers(self):
         return DataTablesHeader(*[DataTablesColumn('Alerts')])
 
     @property
     def rows(self):
         end_date = self.datespan.enddate
-        location = SQLLocation.objects.get(location_id=self.request.GET.get('location_id', ''))
+        sql_location = self.sql_location
         alerts = Alert.objects.filter(
-            location_id__in=location.get_descendants(include_self=True).filter(
-                Q(location_type__administrative=False) | Q(location_id=location.location_id)
+            location_id__in=sql_location.get_descendants(include_self=True).filter(
+                location_type__administrative=False,
+                is_archived=False
```

Comment:
it looks like the previous query was always including the location from the report filter even if it was an administrative location. the new query will filter it out if it's administrative - is that intentional?